### DISCUSSION TITLE
Does networkx support weighted graph conversion to line graphs?

### DISCUSSION THREAD

**Discussion by vymao** [networkx/networkx#4302] [Q&A] [answered]
If I have a weighted, undirected graph, can networkx generate the corresponding line graph with the weights (which would then be node attributes)?

**Comment by dschult** [comment, marked as answer]
```python
G=nx.path_graph(9)
G.add_weighted_edges_from((u, v, 42) for u, v in G.edges)
LG = nx.line_graph(G)
for e in G.edges:
    LG.nodes[e]["weight"] = G.edges[e]["weight"]
```

**Comment by vymao** [comment]
Is the edge attribute of G restricted to be an `int`, or is it general (so we could embed a list as the edge attribute), and would this still work?

**Comment by jarrodmillman** [comment]
You may want to take a look at the tutorial.  In particular, take a look at the section about adding node and edge attributes:
https://networkx.org/documentation/latest/tutorial.html#adding-attributes-to-graphs-nodes-and-edges

If you still have questions, please let us know.
